src/c/urweb.c

```diff
@@ -57,6 +58,50 @@ int uw_really_write(int fd, const void *buf, size_t len) {
   return 0;
 }
 
+int uw_accept(int sockfd, struct sockaddr *addr, socklen_t *addrlen) {
+  static pthread_mutex_t m = PTHREAD_MUTEX_INITIALIZER;
+
+  int ret;
+  int flags;
+
+  flags = fcntl(sockfd, F_GETFL, 0);
+  if(flags < 0)
+    return flags;
+  ret = fcntl(sockfd, F_SETFL, flags | O_NONBLOCK);
+  if(ret < 0)
+    return ret;
+
+  assert(sockfd < FD_SETSIZE);
+  fd_set sockfds;
+  FD_ZERO(&sockfds);
+  FD_SET(sockfd, &sockfds);
+  ret = select(sockfd+1, &sockfds, NULL, NULL, NULL /* wait infinitely */);
+  if(ret < 0)
+    goto unblock;
+  if(ret == 0) {
+    ret = -1;
+    goto unblock;
```

Comment:
Since you’re trying to emulate `accept`, you need to set `errno` and return `-1` whenever there’s an error condition. However, in this case, `select` thinks it returned successfully, so you need to set `errno.` I recommend `errno = EINVAL` (“Invalid argument”); it’s not strictly correct, but it’s one of the allowable `errno` codes from `accept`, and it seems roughly closest to the impossible situation this represents.